src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts

```diff
@@ -34,23 +35,33 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 		@IOpenerService private readonly _openerService: IOpenerService,
 		@INotificationService private readonly _notificationService: INotificationService) {
 		super();
-		this._tryRunTasks();
+		if (this._workspaceTrustManagementService.isWorkspaceTrusted()) {
+			this._tryRunTasks();
+		}
+		this._register(this._workspaceTrustManagementService.onDidChangeTrust(async trusted => {
```

Comment:
Do we still need to listen to this event if the above `isWorkspaceTrusted` check is true?

Reply:
good pt 👍🏼